sdks/python/apache_beam/io/external/xlang_bigqueryio_it_test.py

```diff
@@ -278,7 +278,58 @@ def test_write_with_beam_rows_cdc(self):
           p
           | beam.Create(rows_with_cdc)
           | StorageWriteToBigQuery(
-              table=table_id, cdc_writes_with_primary_key=["name"]))
+              table=table_id,
+              use_at_least_once=True,
+              use_cdc_writes=True,
+              cdc_writes_primary_key=["name"]))
+    hamcrest_assert(p, bq_matcher)
+
+  def test_write_dicts_cdc(self):
+    table = 'write_dicts_cdc'
+    table_id = '{}:{}.{}'.format(self.project, self.dataset_id, table)
+
+    expected_data_on_bq = [
+        # (name, value)
+        {
+            "name": "cdc_test",
+            "value": 5,
+        }
+    ]
+
+    schema = {
+        "fields": [{
+            "name": "name", "type": "STRING"
+        }, {
+            "name": "value", "type": "INTEGER"
+        }]
+    }
+
+    dicts = [{
+        "name": "cdc_test", "value": 3
+    }, {
+        "name": "cdc_test", "value": 5
+    }, {
+        "name": "cdc_test", "value": 4
+    }]
+
+    bq_matcher = BigqueryFullResultMatcher(
+        project=self.project,
+        query="SELECT * FROM {}.{}".format(self.dataset_id, table),
+        data=self.parse_expected_data(expected_data_on_bq))
+
+    with beam.Pipeline(argv=self.args) as p:
+      _ = (
+          p
+          | beam.Create(dicts)
+          | beam.io.WriteToBigQuery(
+              table=table_id,
+              method=beam.io.WriteToBigQuery.Method.STORAGE_WRITE_API,
+              schema=schema,
+              use_at_least_once=True,
+              use_cdc_writes=lambda row: beam.Row(
+                  mutation_type="UPSERT",
+                  change_sequence_number="AAA/" + str(row.value)),
```

Comment:
Note that when the user is writing Python dicts, they should be able to supply a function that works on those dicts (ie. they shouldn't have to know what a Beam Row). Most Python users are not aware that Beam Rows and the Java IO are being used under the hood.